chemprop/v2/tests/unit/test_loss_functions.py

```diff
@@ -0,0 +1,484 @@
+    torch.testing.assert_close(loss, mse)
+
+
+@pytest.mark.parametrize(
+    "preds,targets,mask,w_s,w_t,lt_mask,gt_mask,likelihood",
+    [(
+        torch.tensor([[0, 1]], dtype=torch.float),
+        torch.zeros([1, 1]),
+        torch.ones([1, 2], dtype=torch.bool),
+        torch.ones([1]),
+        torch.ones([2]),
+        torch.zeros([2], dtype=torch.bool),
+        torch.zeros([2], dtype=torch.bool),
+        torch.tensor(0.39894228, dtype=torch.float),
+    )]
+)
+def test_MVE(preds, targets, mask, w_s, w_t, lt_mask, gt_mask, likelihood):
+    """
+    Tests the normal_mve loss function
+    """
+    mve_loss = MVELoss()
+    nll_calc = mve_loss(preds, targets, mask, w_s, w_t, lt_mask, gt_mask)
+    likelihood_calc = np.exp(-1 * nll_calc)
+    torch.testing.assert_close(likelihood_calc, likelihood)
+
+
+@pytest.mark.parametrize(
+    "preds,targets,mask,w_s,w_t,lt_mask,gt_mask,v_kl,expected_loss",
+    [
+        (
+            torch.tensor([[2, 2]]),
+            torch.ones([1, 1]),
+            torch.ones([1, 2], dtype=torch.bool),
+            torch.ones([1]),
+            torch.ones([1]),
+            torch.zeros([1], dtype=torch.bool),
+            torch.zeros([1], dtype=torch.bool),
+            0,
+            torch.tensor(0.6, dtype=torch.float)
+        ),
+        (
+            torch.tensor([[2, 2]]),
+            torch.ones([1, 1]),
+            torch.ones([1, 2], dtype=torch.bool),
+            torch.ones([1]),
+            torch.ones([1]),
+            torch.zeros([1], dtype=torch.bool),
+            torch.zeros([1], dtype=torch.bool),
+            0.2,
+            torch.tensor(0.63862943, dtype=torch.float)
+        )
+    ]
+)
+def test_BinaryDirichlet(preds, targets, mask, w_s, w_t, lt_mask, gt_mask, v_kl, expected_loss):
+    """
+    Test on the dirichlet loss function for classification.
+    Note these values were not hand derived, just testing for
+    dimensional consistency.
+    """
+    binary_dirichlet_loss = BinaryDirichletLoss(v_kl = v_kl)
+    loss = binary_dirichlet_loss(preds, targets, mask, w_s, w_t, lt_mask, gt_mask)
+    print(loss)
+    torch.testing.assert_close(loss, expected_loss)
+
+
+@pytest.mark.parametrize(
+    "preds,targets,mask,w_s,w_t,lt_mask,gt_mask,",
+    [
+        (
+            torch.ones([1, 1]),
+            torch.ones([1, 1]),
+            torch.ones([1, 2], dtype=torch.bool),
+            torch.ones([1]),
+            torch.ones([1]),
+            torch.zeros([1], dtype=torch.bool),
+            torch.zeros([1], dtype=torch.bool),
+        ),
+    ]
+)
+def test_BinaryDirichlet_wrong_dimensions(preds, targets, mask, w_s, w_t, lt_mask, gt_mask):
+    """
+    Test on the dirichlet loss function for classification
+    for dimension errors.
+    """
+    with pytest.raises(RuntimeError):
+        binary_dirichlet_loss = BinaryDirichletLoss()
+        binary_dirichlet_loss(preds, targets, mask, w_s, w_t, lt_mask, gt_mask)

```

Comment:
good test- should we think about catching this and throwing a more descriptive error?

Reply:
I adapted this test from v1. Do we want to print a wrong input dimension error?

Reply:
we decided to ignore my comment